### ISSUE TITLE
Support AirPlay 2 Fast Streaming

### ISSUE THREAD

**Issue by hipwelljo** [AudioKit/AudioKit#1410]
TLDR; It would be fantastic if AudioKit supported the enhanced buffering (aka fast streaming) available using AirPlay 2.

We are working to integrate AirPlay 2 support in our app. We have followed [the documentation](https://developer.apple.com/documentation/avfoundation/airplay_2/getting_airplay_2_into_your_app) and are able to stream the audio to multiple devices simultaneously. The last step is to support fast streaming. Without this, it takes at least a second to begin playing the music on other devices, or a second to pause after tapping the pause button for example. Buffering isn't great. Ultimately it's like AirPlay 1.

The documentation notes two approaches to support this, simple or flexible. Simple would be to use AVPlayer or AVQueuePlayer. AudioKit does not use those so it seems flexible will be the right route, which uses AVSampleBufferAudioRenderer and AVSampleBufferRenderSynchronizer. Here's the link to the [walkthrough article in the docs](https://developer.apple.com/documentation/avfoundation/airplay_2/implementing_flexible_enhanced_buffering_for_your_content). It would be fantastic to see support for that!

Thank you!

**Comment by aure** [comment]
Looks interesting, thanks for the link too.  If you want to make a stab at getting this to work, it would help get the ball rolling. You could even make a pull request if you get close but still have some issues.

**Comment by hipwelljo** [comment]
Unfortunately I don't think I will have the time and resources to implement it given my current workload.

**Comment by aure** [comment]
I gave this a whirl and my feeling is that there's not enough documentation/walkthrough yet to make this actionable.  Perhaps someone will create a github project that does this - basically that's what is needed, a working example (even if it doesn't use AudioKit).
